lib/CGameState.cpp

```diff
@@ -726,7 +726,7 @@ void CGameState::init(const IMapService * mapService, StartInfo * si, bool allow
 	logGlobal->info("\tUsing random seed: %d", si->seedToBeUsed);
 	getRandomGenerator().setSeed(si->seedToBeUsed);
 	scenarioOps = CMemorySerializer::deepCopy(*si).release();
-	initialOpts = CMemorySerializer::deepCopy(*si).release();
+	initialOpts = scenarioOps;
```

Comment:
are you absolutely sure that this is correct? To me it looks that the original intention was to keep these entities completely separate.

Reply:
This is incorrect and will break restart scenario functionality

Reply:
This is incorrect and will break restart scenario functionality

Reply:
this change is incorrect and will break restart scenario functionality

Reply:
@Nordsoft91 fix it here https://github.com/vcmi/vcmi/pull/1111/commits/2c01e9680c3778d5f4a2732eb6f7d01c42fc2501